src/pages/PeoplePage.tsx

```
@@ -0,0 +1,94 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../components/Loader';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import PersonItem from '../components/PersonItem';
+
+const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    if (!isError) {
+      setIsLoading(true);
+      getPeople()
+        .then(result => {
+          const personWithParents = result.map(person => {
+            const modifiedPerson = { ...person };
+            const mam = result.find(p => p.name === person.motherName);
+            const dad = result.find(p => p.name === person.fatherName);
+
+            if (mam) {
+              modifiedPerson.mother = mam;
+            }
+
+            if (dad) {
+              modifiedPerson.father = dad;
+            }
+
+            return modifiedPerson;
+          });
+
+          setPeople(personWithParents);
+        })
+        .catch(() => {
+          setIsError(true);
+          setTimeout(() => {
+            setIsError(false);
+          }, 3000);
```

The current error handling logic hides the error message after 3 seconds. Consider keeping the error message visible until the user takes an action, such as retrying the fetch or dismissing the message manually, to improve user experience.